### ISSUE TITLE
Displaying a table causes it to lose precision

### ISSUE THREAD

**Issue by tom-perkins** [mantidproject/mantid#16465]
*Reported by James Lord in beta testing for 3.7*

#### Expected behaviour
Displaying a TableWorkspace in MantidPlot should not affect the data in it

#### Actual behaviour
Displaying the table causes the values in it to lose precision

#### Steps to reproduce the behaviour
```python
logTab=WorkspaceFactory.createTable()
logTab.addColumn("int","Run")
logTab.addColumn("double","B")

for i in range(77767,77777):
	z=20732.6576324+(i-77767)*19.654535098765
	logTab.addRow((i,z))
AnalysisDataService.addOrReplace("logTab",logTab)

# Full precision
t=mtd["logTab"]
for r in t:
	print r["B"]

# Display the table
tabl = importTableWorkspace("logTab", True)

# Now the values have lost precision
t2 = mtd["logTab"]
for r in t2:
    print r["B"]
```

Output:
20732.6576324
20752.3121675
20771.9667026
20791.6212377
20811.2757728
20830.9303079
20850.584843
20870.2393781
20889.8939132
20909.5484483
20732.7
20752.3
20772.0
20791.6
20811.3
20830.9
20850.6
20870.2
20889.9
20909.5

(note that Python prints 12 significant figures by default)

**Comment by tom-perkins** [comment]
In Mantid 3.5.1, the numbers in the TableWorkspace are not affected (printed the same each time), though the numbers in the displayed table still do have lower precision.

**Comment by tom-perkins** [comment]
Now merged into `release-v3.7` so should be closed
